### PR TITLE
[ADD] l10n_ar_account_withholding: Test ARBA

### PR DESCRIPTION
**Pull request by ica-adhoc**
tickets 50202, 52470

Add button in settings view to test ARBA connection

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://runbot.adhoc.com.ar/ingadhoc/odoo-argentina/pull/619).

File: l10n_ar_account_withholding/wizard/res_config_settings.py [hunk 1]
```diff
@@ -45,3 +46,21 @@ class ResConfigSettings(models.TransientModel):
         'Allow to choose base amount type for withholdings on partners',
         implied_group='l10n_ar_account_withholding.partner_tax_withholding_amount_type',
     )
+
+    def l10n_ar_arba_cit_test(self):
+        self.ensure_one()
+        cuit = self.company_id.partner_id.ensure_vat()
+        _logger.info('Getting ARBA data for cuit %s' % (cuit))
+        try:
+            import pdb; pdb.set_trace()
```
Comment (zaoral) [inline l10n_ar_account_withholding/wizard/res_config_settings.py]: remove import pdb

File: l10n_ar_account_withholding/models/res_company.py [hunk 2]
```diff
@@ -239,3 +238,9 @@ def get_cordoba_data(self, partner, date):
         _logger.info("We've got the following data: \n%s" % data)
 
         return data
+
+    @api.model
+    def _process_message_error(self, ws):
+        message = ws.MensajeError
+        message = message.replace('<![CDATA[', '').replace(']]/>','')
+        raise UserError('No se pudo conectar a ARBA: %s - %s' % (ws.CodigoError, message))
```
Comment (zaoral) [inline l10n_ar_account_withholding/models/res_company.py]: add_() to string "No se pudo conectar a ARBA"

File: l10n_ar_account_withholding/wizard/res_config_settings.py [hunk 3]
```diff
@@ -45,3 +46,21 @@ class ResConfigSettings(models.TransientModel):
...
+            ws.ConsultarContribuyentes(
+                fields.Date.start_of(fields.Date.today(), 'month').strftime('%Y%m%d'),
+                fields.Date.end_of(fields.Date.today(), 'month').strftime('%Y%m%d'),
+                cuit)
+        except Exception as exp:
+            raise UserError('No se pudo conectar a ARBA: %s' % str(exp))
+
+        if ws.CodigoError:
+            self.company_id._process_message_error(ws)
+        raise UserError('La conexión ha sido exitosa')
```
Comment (zaoral) [inline l10n_ar_account_withholding/wizard/res_config_settings.py]: same. add _() at  string 'La conexión ha sido exitosa')

**Review by zaoral — CHANGES_REQUESTED** [review]

**Review by zaoral — APPROVED** [review]

**Comment by zaoral** [comment]
@roboadhoc r+
@roboadhoc override=ci/argentina-ee

**Comment by roboadhoc** [comment]
@zaoral, you may want to rebuild or fix this PR as it has failed CI.

**Comment by roboadhoc** [comment]
I'm sorry, @zaoral. You are not allowed to override this status.

**Comment by nicomacr** [comment]
@roboadhoc override=ci/argentina-ee

**Comment by nicomacr** [comment]
@roboadhoc p=1

**Pull request closed without merging**